src/radical/pilot/pmgr/launching/default.py

```diff
@@ -1037,19 +890,29 @@ def _prepare_pilot(self, resource, rcfg, pilot, expand, tar_name):
                 self._log.error('virtenv_mode is local, no local env found')
                 raise ValueError('no local env found')
 
-        # if cores_per_node is set (!= None), then we need to
-        # allocation full nodes, and thus round up
-        if cores_per_node:
-            cores_per_node = int(cores_per_node)
-            number_cores   = int(cores_per_node *
-                             math.ceil(float(number_cores) / cores_per_node))
+        if number_nodes:
+            number_cores = number_nodes * cores_per_node
```

Comment:
continue from comment above about `cores_per_node` - if default value is used, then requested `number_cores` will be incorrect (for example for Summit, `cores_per_node` will depend on `SMT` level)